apps/coscrad-frontend/src/utils/generic-components/presenters/tables/generic-index-table-presenter/index-table.tsx

```diff
@@ -57,35 +100,101 @@ export const IndexTable = <T extends IBaseViewModel>({
         Object.entries(cellRenderersDefinition) as [keyof T, CellRenderer<T>][]
     );
 
-    return (
-        <div>
-            <h3>{heading}</h3>
-            <div className="records-table">
-                <div>
-                    <table>
-                        <thead>
-                            <tr>
-                                {headingLabels.map(({ headingLabel }) => (
-                                    <th key={headingLabel}>{headingLabel}</th>
+    const table =
+        paginatedData.length === 0 ? (
+            <NotFoundPresenter />
+        ) : (
+            <div>
+                <table>
```

Comment:
We should break the table into a separate component. We might even want to inject the table presenter at some point. this  would allow us to use a list view on mobile, or even potentially use a carousel or some other way of presenting, without rewriting the filtering and pagination.

Reply:
I added a comment to this effect.